src/main/java/org/opensearch/commons/model2/model/Action.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.commons.model2.model;
+
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.commons.model2.AbstractModel;
+import org.opensearch.commons.model2.ToXContentModel;
+import java.util.List;
+
+public class Action extends AbstractModel {
+
+    public static NamedXContentRegistry.Entry XCONTENT_REGISTRY = ToXContentModel.createRegistryEntry(Action.class);
+
+    public String id;
+    public String name;
+    public String destination_id;
+    public Script subject_template;
+    public Script message_template;
+    public boolean throttled_enabled;
+    public Throttle throttle;
+    public ExecutionPolicy action_execution_policy;
+
+    //val throttleEnabled: Boolean,
+    // val throttle: Throttle?,
+    //val id: String = UUIDs.base64UUID(),
+    //val actionExecutionPolicy: ActionExecutionPolicy? = null
+    public Action() {
+        // for serialization
+    }
```

Comment:
I see a few ToDo's across the model classes. Are we planning to update them as part of this PR incrementally ?